torchaudio/functional.py

```diff
@@ -844,6 +844,163 @@ def compute_deltas(specgram, win_length=5, mode="replicate"):
+    for index in range(error.size()[0]):
+        err = error[index]
+        error_offset = torch.cat((torch.zeros(1), err))
+        error[index] = error_offset[:waveform.size()[1]]
+
+    noise_shaped = dithered_waveform + error
+    return noise_shaped
+
+
+@torch.jit.script
+def probability_distribution(waveform, density_function="TPDF"):
+    # type: (Tensor, str) -> Tensor
+    r"""Perform a probability distribution function on a waveform.
+
+    Triangular probability density function (TPDF) dither noise has a 
+    triangular distribution; values in the center of the range have a higher 
+    probability of occurring. 
+
+    Rectangular probability density function (RPDF) dither noise has a 
+    uniform distribution; any value in the specified range has the same
+    probability of occurring.
+
+    Gaussian probability density function (GPDF) has a normal distribution.
+    The relationship of probabilities of results follows a bell-shaped,
+    or Gaussian curve, typical of dither generated by analog sources.
+    Args:
+        waveform (torch.Tensor): Tensor of audio of dimension (channel, time)
+        probability_density_function (string): The density function of a
+           continuous random variable (Default: `TPDF`)
+           Options: Triangular Probability Density Function - `TPDF`
+                    Rectangular Probability Density Function - `RPDF`
+                    Gaussian Probability Density Function - `GPDF`
+    Returns:
+        torch.Tensor: waveform dithered with TPDF
+    """
+    wave_size = waveform.size()[1] - 1
+    number_of_bits = 16
+
+    up_scaling = 2 ** (number_of_bits - 1) - 2
+    signal_scaled = waveform * up_scaling
+    down_scaling = 2 ** (number_of_bits - 1)
+
+    distributed = waveform
+    if (density_function == "RPDF"):
+        random_tensor = int(torch.randint(wave_size, [1, ]).item())
+        RPDF = waveform[0][random_tensor] - 0.5
+
+        signal_scaled_RPDF = signal_scaled + RPDF
+        quantised_signal_scaled_RPDF = torch.round(signal_scaled_RPDF)
+        quantised_signal_RPDF = quantised_signal_scaled_RPDF / down_scaling
+
+        distributed = quantised_signal_RPDF
+    elif (density_function == "GPDF"):
+        gaussian = torch.distributions.normal.Normal(torch.mean(waveform), 1).sample()
+
+        signal_scaled_gaussian = signal_scaled + gaussian
+        quantised_signal_scaled_gaussian = torch.round(signal_scaled_gaussian)
+        quantised_signal_gaussian = quantised_signal_scaled_gaussian / down_scaling
+
+        distributed = quantised_signal_gaussian
+    else:
+        TPDF = torch.bartlett_window(wave_size + 1)
+
+        signal_scaled_TPDF = signal_scaled
+        for index in range(signal_scaled_TPDF.size()[0]):
+            signal_scaled_TPDF[index] += TPDF
+
+        quantised_signal_scaled_TPDF = torch.round(signal_scaled_TPDF)
+        quantised_signal_TPDF = quantised_signal_scaled_TPDF / down_scaling
```

Comment:
I see these lines with `round` and `/ down_scaling` repeated. Would it be cleaner to move them out of the conditional?